### PR TITLE
Changes in the backend code to comply to PEP8 + refactor spacy_matcher.py function

### PR DESCRIPTION
**Pull request by ANA-POTJE**
Following the code review made by @dh7892 (#91):

1. Pulled @dh7892 changes into a branch;
2. Refactored the function spacy_matcher.py, removed the input parameter "tags" and included list comprehension command to improve the list of dictionaries creation (output);
3. Followed PEP8 guidelines and renamed variables and functions;
4. Followed PEP8 guidelines for formatting the code: docstrings, indentation and spacing (identified with Pylint);
5. Fixed the Pytest modules to reflect the changes;
6. Updated README.md file with the changes

### CODE REVIEW

File: src/download_punkt.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import nltk
```
Comment (jsjohnstone) [inline src/download_punkt.py:1]: I should add some comments to this file too.

Comment (jsjohnstone) [inline src/download_punkt.py:1]: I should add some comments to this file.

File: src/lambda_function.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
 from aws_xray_sdk.core import xray_recorder
 from aws_xray_sdk.core import patch
 import json
-import getCredibilityScore as cr
-import sentimentAnalysis as sa
-import getBiasScore as bs
-import spacyMatcher as sm
+import get_credibility_score as cr
```
Comment (dh7892) [inline src/lambda_function.py:4]: I'm not a massive fan of using these "nicknames" for things. What's wrong with just importing `get_credibility_score` and calling it by that name when you use it?

File: src/lambda_function.py [hunk 3]
```diff
@@ -15,136 +15,186 @@
...
         url = event['url']
     except KeyError:
         logger.info(f'LambdaFunction: URL is not present.')
-        return {"error" : "No URL provided"}
+        return {"error": "No URL provided"}
 
     xray_recorder.end_subsegment()
 
-    #### Adding "https://" to the URL if not present
+    """
```
Comment (dh7892) [inline src/lambda_function.py:36]: This was a single line comment and not a docstring so it's fine to use `#` It looks a bit odd to have it as a string comment. But maybe you could have just begun with a single `#`

File: src/lambda_function.py [hunk 4]
```diff
@@ -15,136 +15,186 @@
...
+                              'outcome': {"error": 'The credibility score '
+                                                   'was not available.'}}
+        object['results'].append(credibility_result)
+
+    """
+    SentimentAnalisys
+    """
     logger.info(f'LambdaFunction: Trying to get sentimentAnalysis score...')
-    sentanalysisresult = {}
+    sentresult = {}
```
Comment (dh7892) [inline src/lambda_function.py:82]: `sent_result` ?

File: src/lambda_function.py [hunk 5]
```diff
@@ -15,136 +15,186 @@
...
-            logger.info(sentanalysisresult)
-            object['article'] = {'error': "The article summary could not be generated"}
-            object['results'].append({'type': 'polarity',     "outcome": {"error" : "The polarity score could not be calculated."}})
-            object['results'].append({'type': 'objectivity', "outcome": {"error" : "The objectivity score could not be calculated."}})
+        sentresult = sa.sentiment_analysis(url)
+        if sentresult['text'] == '-1':
+            logger.info(f'LambdaFunction: sentimentAnalysis returned -1, '
+                        'dumping:')
+            logger.info(sentresult)
+            object['article'] = {'error':
```
Comment (dh7892) [inline src/lambda_function.py:89]: It looks like you end up creating a similar object dictionary in a few different places and it looks pretty bulky having it inside this if statement. Perhaps you could create the error dict somewhere else in this file and then just do something like: `object['results] = failed_polarity_result` here?

File: src/lambda_function.py [hunk 6]
```diff
@@ -15,136 +15,186 @@
...
-        object['article'] = {'error': "The article summary could not be generated"}
-        object['results'].append({'type': 'polarity',     "outcome": {"error" : "The polarity score could not be calculated."}})
-        object['results'].append({'type': 'objectivity', "outcome": {"error" : "The objectivity score could not be calculated."}})
-
-
-    # BiasScore
+        sentresult = {'text': -1}
+        object['article'] = {'error': 'The article summary could not be '
+                                      'generated'}
+        object['results'].append({'type': 'polarity',
```
Comment (dh7892) [inline src/lambda_function.py:127]: Same as above in terms of the dict

File: src/spacy_matcher.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+from aws_xray_sdk.core import xray_recorder
+import spacy
+
+
+@xray_recorder.capture('spacy_matcher')
+def spacy_matcher(text):
```
Comment (dh7892) [inline src/spacy_matcher.py:6]: This is a lot tidier now!

**Review by dh7892 — COMMENTED** [review]
A few points you might like to consider (not saying you should do these now!)

* You don't have to make all comments use the """ style. It's normally for the docstrings (the special comments inside modules, classes and functions) or for comments that span many lines. Normal comments that are only a few lines look better as `#` comments
* I like my tests to be commented because I think they can sometimes be very subtle in what they are doing so it's hard for someone to come back later to refactor them. We use "gherkin" syntax for this: "GIVEN I am logged in as a normal user, WHEN I trigger some event, THEN I expect this response" This format is very nice for making it clear your predicates, actions and expected results.
* If you haven't heard of it already, I recommend Black. It's a Python auto-formatter. It will save you from having to manually reformat stuff like spacing and line breaks.

Hope this is helpful.

**Review by jsjohnstone — APPROVED** [review]

**Comment by ANA-POTJE** [comment]
Hi @dh7892, thanks very much for your comments.
We are merging the changes implemented so far, will improve the code including your suggestions at one point soon ;)
Many thanks!

**Pull request merged**